src/resources/log-message.ts

```diff
@@ -0,0 +1,70 @@
+import "core-js/modules/web.url.can-parse";
+import { fromError } from "stacktrace-js";
+import { UAParser } from "ua-parser-js";
+
+// URL paths to remove from filenames and max stack trace lines for brevity
+const REMOVAL_PATHS =
+  /^\/(?:home-assistant\/frontend\/[^/]+|unknown|\/{2}\.)\//;
+const MAX_STACK_FRAMES = 10;
+
+export const createLogMessage = async (
+  error: unknown,
+  intro?: string,
+  messageFallback?: string,
+  stackFallback?: string
+) => {
+  const lines: (string | undefined)[] = [];
+  // Append the originating browser/OS to any intro for easier identification
+  if (intro) {
+    const parser = new UAParser();
+    const {
+      name: browserName = "unknown browser",
+      version: browserVersion = "",
+    } = parser.getBrowser();
+    const { name: osName = "unknown OS", version: osVersion = "" } =
+      parser.getOS();
+    const browser = `${browserName} ${browserVersion}`.trimEnd();
+    const os = `${osName} ${osVersion}`.trimEnd();
```

Comment:
Do we need to use `trimEnd` here, or can we just use `trim`?

Reply:
`trim` should be fine too - I'll change that.